MAUI/DataForm/grouping.md

```diff
@@ -0,0 +1,386 @@
+    public string FirstName { get; set; }
+
+    [Display(GroupName = "Name")]
+    [DataFormDisplayOptions(RowOrder = 0, ItemsOrderInRow = 2)]
+    public string LastName { get; set; }
+
+    [Display(GroupName = "Name")]
+    [DataFormDisplayOptions(RowOrder = 0, ItemsOrderInRow = 1)]
+    public string MiddleName { get; set; }
+
+{% endhighlight %}
+{% highlight C# %}
+
+        if (e.DataFormGroupItem != null)
+        {
+            if (e.DataFormGroupItem.GroupName == "Name")
+            {
+                e.DataFormGroupItem.ColumnCount = 3;
+            }
+        }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing the group name for the editors
+
+The `GroupName` for the group can be changed in the `GenerateDataFormItem` event.
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+       private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem != null)
+            {
+                if (e.DataFormItem.GroupName == "Name")
+                {
+                    e.DataFormItem.GroupName = "Name Group";
+                }
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing the layout for the group
+
+The linear or grid layout for the particular group can be loaded by handling the `GenerateDataFormItem` event.
+By setting the `ColumnCount` property in the data form, non-grouped items only will be arranged in the grid layout. To load the grid layout, set the [ColumnCount] for the `DataFormGroupItem`
+
+{% tabs %}
+{% highlight C# %}
+
+public class ContactsInfo
+{
+    [Display(GroupName = "Name")]
+    public string FirstName { get; set; }
+
+    [Display(GroupName = "Name")]
+    public string LastName { get; set; }
+
+    [Display(GroupName = "Name")]
+    public string MiddleName { get; set; }
+
+    [Display(GroupName = "Details")]
+    public string Address { get; set; }
+
+    [Display(GroupName = "Details")]
+    public double PhoneNumber { get; set; }
+}
+
+{% endhighlight %}
+{% highlight C# %}
+
+this.dataForm.DataObject = new ContactsInfo();
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormGroupItem != null)
+        {
+            if (e.DataFormGroupItem.GroupName == "Name")
+            {
+                e.DataFormGroupItem.ColumnCount = 3;
+            }
+            else if (e.DataFormGroupItem.GroupName == "Details")
+            {
+                e.DataFormGroupItem.ColumnCount = 2;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Loading the group in collapsed state
+
+By default, the group will be loaded in expanded state. You can collapse the group by setting the [IsExpanded] property of [DataFormGroupItem] class to `false`.
```

Comment:
By default, the group will be loaded in the expanded state. Collapse the group by setting the [IsExpanded] property of the [DataFormGroupItem] class to `false.`

Reply:
Content changed